lua/spectre_common.lua

```diff
@@ -21,6 +21,7 @@ local HEADERS = {
     CACHE_STATUS = 'Spectre-Cache-Status',
     B3_TRACEID = 'X-B3-TraceId',
     ZIPKIN_ID = 'X-Zipkin-Id',
+    DOWNSTREAM_ERROR = 'X-Downstream-Error'
```

Comment:
Seems this header reflects downstream status that is not always error?
For example, statuses like 3xx, 4xx

Reply:
Yes 4xx is intentional, but good point regarding 3xx, I will adjust this accordingly

Reply:
I feel this might be a good thing, if we log all original status codes, with cache status, then we could assert via an alert that we never reach unexpected states in production (example: never seeing a combination of 3xx && cached)

Reply:
@thebostik agreed, will take that approach.